interface/forms/gad7/report-as-pdf.php

```diff
@@ -0,0 +1,145 @@
+// include scores_array and total for backward compatibility
+            if ($key == "id" || $key == "pid" || $key == "user" || $key == "groupname" || $key == "authorized" || $key ==  "activity" || $key == "date" || $value == "" || $key == "scores_array" || $key == "total" || $value == "0000-00-00 00:00:00") {
+                continue;
+            }
+            if ($key !== "difficulty") {
+                if (is_numeric($value)) {
+                    $gad7_total += $value;
+                }
+            }
+        // find score description from the total
+            switch (intdiv($gad7_total, 5)) {
+                case 0:
+                    $exp = $str_score_analysis[0];
+                    break;
+                case 1:
+                case 2:
+                    $exp = $str_score_analysis[5];
+                    break;
+                default:
+                    $exp = $str_score_analysis[15];
+            }
+        }
+    } else {
+        echo "<br>No form data</br>";
+    }
+ /* generate a pdf version of the form,   int pid - patient id , int id - gad7 form number
+ */
+    $config_mpdf = array(
+        'tempDir' => $GLOBALS['MPDF_WRITE_DIR'],
+        'mode' => $GLOBALS['pdf_language'],
+        'format' => $GLOBALS['pdf_size'],
+        'default_font_size' => '9',
+        'default_font' => 'dejavusans',
+        'margin_left' => $GLOBALS['pdf_left_margin'],
+        'margin_right' => $GLOBALS['pdf_right_margin'],
+        'margin_top' => $GLOBALS['pdf_top_margin'],
+        'margin_bottom' => $GLOBALS['pdf_bottom_margin'],
+        'margin_header' => '',
+        'margin_footer' => '',
+        'orientation' => $GLOBALS['pdf_layout'],
+        'shrink_tables_to_fit' => 1,
+        'use_kwt' => true,
+        'autoScriptToLang' => true,
+        'keep_table_proportions' => true,
+        'debug' => true,
+        'allow_output_buffering' => false
+        );
+    $mpdf = new mPDF($config_mpdf);
+
+    /* reformat the date from the NOW() format generated by formSubmit, which is Ymd, into configured format plus time */
+    $date_time = explode(" ", $data['date'], 2); /* split into date and time */
+    $date_elements = explode('-', $date_time[0], 3); /*split date into y, m, d */
+    switch ($GLOBALS['date_display_format']) {
+        case 1: /*mm-dd-yyy*/
+            $formated_date = $date_elements[1] . '-' . $date_elements[2] . '-' . $date_elements[0];
+            break;
+        case 2:  /*dd-mm-yyy */
+            $formated_date = $date_elements[2] . '-' . $date_elements[1] . '-' . $date_elements[0];
+            break;
+     /* if 0 - YYYYMMDD nothing to do */
+        default:
+            $formated_date = $date_time[0];
+    }
+    $client_name = getPatientNameFirstLast($pid);
+  /* strip out white space - replace with underscore  - for use in file name*/
+    $client_name_fn = str_replace(" ", "_", $client_name);
+    $client_file_name = $client_name_fn . '_gad7_' . $id . '.pdf' ; /* downloaded file name */
```

Comment:
to prevent directory walking vulnerability:
```
$client_file_name = $client_name_fn . '_gad7_' . check_file_dir_name($id) . '.pdf' ; /* downloaded file name */
```